### DISCUSSION TITLE
Comment/Uncomment one ore multiple lines?

### DISCUSSION THREAD

**Discussion by manuschillerdev** [LazyVim/LazyVim#690] [Q&A] [answered]
I feel stupid for asking this question, but I am not able to comment/uncomment lines with nvim/lazyvim.
What would be the right approach to toggle a commented out line - independent of the current programming language used in the file?

**Comment by dpetka2001** [comment]
Lazyvim provides a keymap for searching available keymaps `<space>sk`. If you type comment it will show you available keymaps for comment commands. By default Lazyvim uses `mini.comment` which uses `gcc` for commenting/uncommenting a line.

**Reply by manuschillerdev** [reply]
<space>sk is a goldmine, thanks!

If I do `<space>sk`, search for `comment`, highlight the entry "Comment line" and press Enter, it works as intended.
However, if I try `:gcc` or `<space>gcc` (I'm in normal mode) it says command not found/enters my git commits.

Any idea which beginner mistake I am making now? :) Thanks for your help!

**Reply by dpetka2001** [reply]
While in normal mode just press `gcc`. No need to press `:` in normal mode. `:` opens the command line pop-up menu.

**Reply by dpetka2001** [reply, marked as answer]
The command for commenting is plain `gcc`. It doesn't contain `<space>`. `<space>` is only needed for commands that do have it. Commenting a line doesn't have it. That's why in `<space>sk` it shows it only as `gcc`. If you search for "Key maps" you will see that it's mapped to `<space>sk` that's why you need to press it in that case.

**Comment by metal3d** [comment]
One remark, because I use it a lot: `gc%` => (un)comments the block

**Comment by lupodellasleppa** [comment]
why if I type `gc` really fast it seems to register `c` only, while typing `g`, waiting for a moment, then typing `c` actually brings up the comment menu? It happens to me a lot to `gcc` very fast with the result being I replace my whole line. It's quite weird having to wait that moment after `g` is typed

**Reply by fmorroni** [reply]
I also have this problem. I created this discussion https://github.com/LazyVim/LazyVim/discussions/7241 to see if I can find a solution.

**Comment by warren124bush** [comment]
Never feel stupid—LazyVim makes it simple! Just use `gcc` to toggle the current line or `gc` in visual mode. It handles language-specific syntax automatically using Treesitter.
